app/Models/User.php

```diff
@@ -951,12 +951,12 @@ public function groupIds()
 
     public function findUserGroup($group, $activeOnly)
     {
-        $groupId = $group->getKey();
+        $byGroupId = $this->memoize(__FUNCTION__.':byGroupId', fn () => $this->userGroups->keyBy('group_id'));
 
-        foreach ($this->userGroups as $userGroup) {
-            if ($userGroup->group_id === $groupId) {
-                return $activeOnly && $userGroup->user_pending ? null : $userGroup;
-            }
+        $userGroup = $byGroupId->get($group->getKey());
+
+        if ($userGroup !== null && (!$activeOnly || !$userGroup->user_pending)) {
```

Comment:
I wonder if inverting the condition to `!($activeOnly && $userGroup->user_pending)` or 
```php
if ($userGroup === null || ($activeOnly && $userGroup->user_pending)) {
    return null;
}
```
would be easier to read?